MaxText/train.py

```diff
@@ -279,6 +279,8 @@ def train_loop(config, state=None):
 
   start_step = get_first_step(state) # this is the start_step for training
   first_profiling_step = start_step + config.skip_first_n_steps_for_profiler
+  last_profiling_step = np.min([first_profiling_step + config.profiler_steps - 1, config.steps - 1])
```

Comment:
please use https://numpy.org/doc/stable/reference/generated/numpy.clip.html

Reply:
Changed, glad to learn of numpy clip. Although numpy.clip uses min(a_max, max(a,min_bound)) which is the opposite order of what I had before. This changes the behavior of the edge case when first_profiling_step >= config.steps (e.g. steps = 1) - so I've added an assert for it